sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { toBase64 } from '@mysten/bcs';
+import { secp256r1 } from '@noble/curves/p256';
+import { blake2b } from '@noble/hashes/blake2b';
+import { randomBytes } from '@noble/hashes/utils';
+import type {
+	AuthenticationCredential,
+	RegistrationCredential,
+} from '@simplewebauthn/typescript-types';
+
+import { PasskeyAuthenticator } from '../../bcs/bcs.js';
+import type { IntentScope, SignatureWithBytes } from '../../cryptography/index.js';
+import { messageWithIntent, SIGNATURE_SCHEME_TO_FLAG, Signer } from '../../cryptography/index.js';
+import type { PublicKey } from '../../cryptography/publickey.js';
+import type { SignatureScheme } from '../../cryptography/signature-scheme.js';
+import {
+	parseDerSPKI,
+	PASSKEY_PUBLIC_KEY_SIZE,
+	PASSKEY_SIGNATURE_SIZE,
+	PasskeyPublicKey,
+} from './publickey.js';
+
+export interface PasskeyCreateOptions {
+	timeout: number; // The timeout for the passkey creation dialog.
+	displayName: string; // The display name of the user, this is shown at creation dialog
+	rpName: string; // The name of the relying party, this is shown to user in the passkey creation dialog.
+}
+
+export interface PasskeySignOptions {
+	timeout: number; // The timeout for the passkey authentication dialog.
+}
+
+function validateCreateOptions(options: PasskeyCreateOptions) {
+	if (options.timeout <= 0) throw new Error('Timeout must be positive');
+	if (!options.displayName?.trim()) throw new Error('Display name cannot be empty');
+	if (!options.rpName?.trim()) throw new Error('RP name cannot be empty');
+	if (options.timeout > 300000) throw new Error('Timeout cannot exceed 5 minutes');
+}
+
+export interface PasskeyProvider {
+	create(options: PasskeyCreateOptions): Promise<RegistrationCredential>;
+	get(challenge: Uint8Array, timeout: number): Promise<AuthenticationCredential>;
+}
+
+// Default browser implementation
+export class BrowserPasskeyProvider implements PasskeyProvider {
+	async create(options: PasskeyCreateOptions): Promise<RegistrationCredential> {
+		return (await navigator.credentials.create({
+			publicKey: {
+				challenge: new TextEncoder().encode('Create passkey wallet on Sui'),
+				rp: {
+					name: options.rpName,
+				},
+				user: {
+					id: randomBytes(10),
+					name: 'sui-wallet-user',
```

Comment:
Is this name required?

Reply:
yeah its required, error if not provided

```
index.ts:87 Uncaught (in promise) TypeError: Failed to execute 'create' on 'CredentialsContainer': Failed to read the 'publicKey' property from 'CredentialCreationOptions': Failed to read the 'user' property from 'PublicKeyCredentialCreationOptions': Failed to read the 'name' property from 'PublicKeyCredentialEntity': Required member is undefined.
```

although this value is not visible to the user or used for anything, since the id is random is sufficient. is there a preferred default value i can put here?